doc/design/bucketpath.rst

```diff
@@ -0,0 +1,157 @@
+==============================
+Design Document Bucket Path(s)
+==============================
+
+Introduction
+============
+
+- Purpose of the document / Problem description
+- Definitions and Acronyms
+
+Overview
+========
+
+Design Goals
+============
+
+Path Design
+===========
+
+Future Considerations
+=====================
+
+Sudo Code Outline
+=================
+
+.. code-block:: python
+
+    from exasol.bucketfs import PathBuilder
+
+    Path = PathBuilder(credentials)
```

Comment:
is this only for serialization or is this the factory?

Reply:
This is the factory, it is supposed to bring together the serialization `as_uri()` and the credentials (and/or other outside information)